plugins/discourse-presence/assets/javascripts/discourse/components/composer-presence-display.js.es6

```diff
@@ -15,115 +14,68 @@ export default Component.extend({
   topic: null,
   reply: null,
   title: null,
+  isWhispering: null,
 
-  // Internal variables
-  previousState: null,
-  currentState: null,
-  presenceUsers: null,
-  channel: null,
-
+  presenceManager: readOnly("topic.presenceManager"),
+  users: readOnly("presenceManager.users"),
+  editingUsers: readOnly("presenceManager.editingUsers"),
   isReply: equal("action", "reply"),
-  shouldDisplay: gt("users.length", 0),
 
   @on("didInsertElement")
-  composerOpened() {
-    this._lastPublish = new Date();
-    once(this, "updateState");
-  },
-
-  @observes("action", "post.id", "topic.id")
-  composerStateChanged() {
-    once(this, "updateState");
+  subscribe() {
+    const presenceManager = this.presenceManager;
```

Comment:
I would just use this.presenceManager directly, no need for this variable IMO

Reply:
o good catch :+1: